### DISCUSSION TITLE
Suddenly "Outdated Client"

### DISCUSSION THREAD

**Discussion by braddo99** [itzg/docker-minecraft-bedrock-server#244] [General]
Get message "outdated client" when attempting to connect to my server. 1000 posts on the web say this is because, duh, the client is outdated, but also that it should be automatically updated on Windows (11).

So, I go through the rigamarole of finding my right windows login and reinstall with the latest version. MS does not make this easy! However, same error. Updated the latest bedrock server image here, and still same condition. Anyone have any other troubleshooting tips?

**Comment by itzg** [comment]
It's a (newly) known issue that has been caused by Mojang's release publishing.

https://github.com/itzg/docker-minecraft-bedrock-server/issues/243

**Comment by esarella** [comment]
> Get message "outdated client" when attempting to connect to my server. 1000 posts on the web say this is because, duh, the client is outdated, but also that it should be automatically updated on Windows (11).
> 
> So, I go through the rigamarole of finding my right windows login and reinstall with the latest version. MS does not make this easy! However, same error. Updated the latest bedrock server image here, and still same condition. Anyone have any other troubleshooting tips?

you can fix it specifying the version 

docker run -d -it --name bds-flat-creative \
  -e EULA=TRUE -e LEVEL_TYPE=flat -e GAMEMODE=creative \
  -p 19132:19132/udp itzg/minecraft-bedrock-server  VERSION=1.7.9
  
  or if you're using docker-compose specify VERSION in your docker-compose.yml

**Comment by csellers85** [comment]
I tried to the do the a rollback to the "previous" version but haven't had any luck. I've even gone as far as to create a new container. Am I doing something wrong? I've tried in both in CLI and using Portainer.

**Comment by itzg** [comment]
@csellers85 this just now worked for me:

```
docker run -it --rm -e EULA=true -p 19132:19132/udp -e VERSION=PREVIOUS itzg/minecraft-bedrock-server
```

**Comment by csellers85** [comment]
Thanks for the clarification, I found in portainer where to edit that variable so it would pull the previous version.

**Comment by braddo99** [comment]
Hello, I am once again getting "outdated client" message. Previously I was able to specify a particular dot release and get the connection going. However, this afternoon once again I get the error with the same server that was previously working. I guess the client was updated again and again a mismatch.

I did get the latest container from dockerhub and switch my make file to version=Previous and version=Latest, and neither of these worked.

**Reply by itzg** [reply]
I would suggest getting the version the client reports, looking up, such as on Fandom, the bedrock server version that goes with that, and then set VERSION specifically with that version.

**Reply by braddo99** [reply]
Anyone know how to get that version? I looked over on Fandom, but couldn't figure how to locate. Is it possible to get it from an info screen somewhere or a / command on the console in Minecraft client? Thanks, the server has been awesome and fun to play on for myself and kids inside our network!

**Reply by itzg** [reply]
I'm a bit confused. The startup automatically gets that latest version. If you want to see specifically what it used then it is reported in the logs.

**Reply by braddo99** [reply]
Sorry for being dense. After working for a while, Minecraft for Windows again updated, and again I cannot connect to a world in my ITZG container. The error is "cannot connect to outdated server". So, the "latest" image is serving up a version that is outdated according to the Minecraft client. I'm not sure but I think the automatically latest version on startup might only apply to non-bedrock servers? At least that might be implied by the readme. You mentioned logs - I don't see any logs in my host machine directory dedicated to this container (volume) and if I do docker exec -it bash and look around, there are no *.log files. I did also look up a server.properties flag that should maybe make some log files, "content-log-file-enabled=true" but for whatever reason the container failed to start with this property set.

**Reply by DominusExult** [reply]
I just needed to restart my container today to get the latest that Minecraft can connect to again.
In Docker GUI you should be able to see the logs of the container.

**Reply by itzg** [reply]
Good point @DominusExult .

The auto-version lookup is only performed at container startup, so it is required to restart the container to have it check for and download updates.

On a non-Desktop system that would be

```
docker restart <container name or ID>
```

BTW, to see the container logs use
```
docker logs <container name or ID>
```
